api/src/libs/profile.ts

```diff
@@ -29,12 +29,19 @@ const { NamespaceModel, ClusterModel, ProfileModel, ContactModel } = dm;
 
 export const getProvisionStatus = async (profile: ProjectProfile): Promise<boolean> => {
   try {
-    const primaryCluster: Cluster = await ClusterModel.findByName(profile.primaryClusterName);
-    if (!primaryCluster.id || !profile.id) {
-      throw new Error('Unable to get primary cluster id or profile id');
+    const clusters = await getClusters(profile)
+    for (const cluster of clusters) {
+      const isClusterProvisioned = await NamespaceModel.getProjectSetProvisionStatus(
+        Number(profile.id),
+        Number(cluster.id)
+      );
+
+      if (!isClusterProvisioned){
```

Comment:
Can you just `return isClusterProvisioned`? or `return isClusterProvisioned ? true : false`.